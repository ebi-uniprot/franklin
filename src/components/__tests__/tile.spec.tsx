import { fireEvent, render } from '@testing-library/react';

import Tile from '../tile';
import { ProtVistaIcon } from '..';

import colors from '../../styles/colours';

let component: ReturnType<typeof render>;
const clickFn = jest.fn();

describe('Tile component', () => {
  beforeEach(() => {
    component = render(
      <Tile
        title="Tile title"
        subtitle="Subtitle"
        description="My description"
        backgroundImage={ProtVistaIcon}
        backgroundColor={colors.colourUniprotkb}
        onClick={clickFn}
        gradient
      />
    );
  });

  test('should render', () => {
    const { asFragment } = component;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should handle click', () => {
    const { getByText } = component;
    fireEvent.click(getByText('Subtitle'));
    expect(clickFn).toHaveBeenCalled();
  });

  test('should render with default specified width', () => {
    const { asFragment } = render(
      <Tile title="Tile title" onClick={clickFn} width="20rem" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
