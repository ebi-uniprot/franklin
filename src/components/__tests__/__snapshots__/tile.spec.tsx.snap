// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component should render 1`] = `
<DocumentFragment>
  <div
    class="tile tile-gradient"
    style="--tile-background: #00639a; width: 100%;"
  >
    <div
      aria-hidden="true"
      class="tile__background-image"
    >
      test-file-stub
    </div>
    <a
      class="tile__main-content"
      href="/"
    >
      <span>
        <h3
          class="tile__header medium"
        >
          Tile title
        </h3>
        <h4
          class="tile__subtitle tiny"
        >
          Subtitle
        </h4>
      </span>
    </a>
    <small
      class="tile__description"
    >
      My description
    </small>
  </div>
</DocumentFragment>
`;

exports[`Tile component should render with default specified width 1`] = `
<DocumentFragment>
  <div
    class="tile"
    style="width: 20rem;"
  >
    <div
      aria-hidden="true"
      class="tile__background-image"
    />
    <a
      class="tile__main-content"
      href="/"
    >
      <span>
        <h3
          class="tile__header medium"
        >
          Tile title
        </h3>
      </span>
    </a>
  </div>
</DocumentFragment>
`;
